### ISSUE TITLE
Incorrect rendering of internal links with dots

### ISSUE THREAD

**Issue by tevino** [gollum/gollum#1570]
**Describe the bug**
The part after the `.` in the internal links are ignored in the rendered HTML.

**TLDR:** `[[this.link]]` is treated like `[[this]]` and finally becomes:

```<a href="/this.link.md">this</a>```

**For example**

`[[v1.2]]` is a link to an **existing** file `v1.2.md`, the link works, but it is displayed as `v1` that's  
```HTML
<a class="internal present" href="/v1.2.md">v1</a>
```
in HTML.

Adding `.md` to the link could work around this, but it doesn't seem consistent. I mean, if the `herf` is `X.md` the link should be displayed as `X`.

**To Reproduce**
1. Create a page of filename `v1.2.md`
2. Create another page with an internal link ``[[v1.2]]`` in its content
3. Save then view the second page in rendered mode

**Expected behavior**
The link displayed as `v1.2`, that's 
```HTML
<a class="internal present" href="/v1.2.md">v1.2</a>
```
in HTML.

**Environment Info**

```shell
> gollum --versions
Gollum 5.0.1
Running on: universal.x86_64-darwin19 with Ruby version 2.6.3
Using:
rugged 0.99.0
gollum-rugged_adapter 1.0
gollum-lib 5.0.4
Markdown rendering gem: kramdown
Other renderering gems:
none
```

**Comment by tevino** [comment]
@dometto Any idea about this?

**Comment by dometto** [comment]
@tevino see https://github.com/gollum/gollum-lib/pull/381

**Comment by dometto** [comment]
Will be fixed when the changes in https://github.com/gollum/gollum-lib/pull/381 are released

**Comment by tevino** [comment]
Thanks for the patch @dometto
